src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,136 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        putValue(newNode);
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+        int position = 0;
+        if (key != null) {
+            position = Math.abs(hash(key) % capacity);
+        }
+        if (key == null || (table[position].next == null && key.equals(table[position].key))) {
+            return table[position].value;
+        } else {
+            Node<K, V> tempNode = table[position];
+            while (tempNode != null) {
+                if (key.equals(tempNode.key)) {
+                    return tempNode.value;
+                }
+                tempNode = tempNode.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void resize() {
+        Node<K, V>[] oldTable = table;
+        if (oldTable == null) {
+            capacity = DEFAULT_INITIAL_CAPACITY;
+            threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+            table = (Node<K, V>[]) new Node[capacity];
+        } else {
+            capacity = capacity * 2;
+            threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+            table = (Node<K, V>[]) new Node[capacity];
+            size = 0;
+            for (int j = 0; j < oldTable.length; j++) {
+                Node<K, V> tempNode = oldTable[j];
+                if (tempNode != null) {
+                    putValue(tempNode);
+                    if (tempNode.next != null) {
+                        tempNode = tempNode.next;
+                        do {
+                            putValue(tempNode);
+                            tempNode = tempNode.next;
+                        } while (tempNode != null);
+                    }
+                }
+            }
+        }
+    }
+
+    public void putValue(Node<K, V> node) {
```

Comment:
```suggestion
    private void putValue(Node<K, V> node) {
```

and remember,  that private methods should follow after public